R/ExperimentHub-class.R

```diff
@@ -92,8 +92,17 @@ setMethod("cache", "ExperimentHub",
     ## Try to install/load; don't fail if not available
     success <- paste0("see ?", pkg, " and browseVignettes('",
                       pkg, "') for documentation")
-    if (!pkg %in% rownames(installed.packages()))
-        BiocManager::install(pkg, suppressUpdates=TRUE)
+    nopkg <- paste0(pkg,
+                    " not installed.\n  Full functionality, documentation, and loading of data might not be possible without installing")
+    if (!pkg %in% rownames(installed.packages())){
+        message(nopkg)
+        if (interactive()){
+            txt <- paste0("Install ", pkg, " (yes/no): ")
+            response <- substr(tolower(readline(txt)), 1, 1)
+            doit <- switch(response, y = TRUE, n = FALSE, FALSE)
+            if (doit) BiocManager::install(pkg, suppressUpdates=TRUE)
```

Comment:
Maybe re-use `AnnotationHub:::.ask()` or pull out this in a reusable function

```{r}
## return TRUE when user replies with 'yes' to prompt, FALSE when 'no'
askUserYesNo <-
    function(prompt, interactive.only = TRUE)
{
    if (interactive.only && !interactive())
        return(FALSE)
    responses = c("yes", "no")
    msg1 <- paste0(prompt, " [", paste(responses, collapse = ", "), "]? ")
    msg2 <- paste0("reply with '", paste(responses, collapse = "' or '"), "'")
    repeat {
        userResponse <- trimws(tolower(readline(msg1)))
        if (userResponse %in% responses)
            break
        message(msg2)
    }
    identical(userResponse, responses[1])
}
```
and then use is

```{r}
        if (askUserYesNo(paste0("install '", pkg, "'")))
        ## if (interactive() && identical(AnnotationHub:::.ask(...), "yes"))
            BiocManager::install(pkg, suppressUpdates=TRUE)
```

This functionality might be a candidate for BiocBaseUtils @LiNk-NY